docs/topics/arrays.md

```diff
@@ -31,32 +50,244 @@ fun main() {
+
+To return the sum of all elements in an array, use the [sum()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/sum.html) function.
+
+```Kotlin
+fun main() { 
+// sampleStart
+    val sumArray = arrayOf(1, 2, 3)
+
+    println(sumArray.sum()) 
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="sum-array-kotlin"}
+
+> The `sum()` function can only be used with arrays of numeric data types.
+>
+{type="note"}
+
+### Shuffle()
+
+To randomly shuffle the elements in an array, use the [shuffle()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/shuffle.html) function.
+
+```Kotlin
+fun main() {
+// sampleStart
+    val simpleArray = arrayOf(1, 2, 3)
+
+    simpleArray.shuffle()
+    println(simpleArray.joinToString())
+
+    simpleArray.shuffle()
+    println(simpleArray.joinToString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="shuffle-array-kotlin"}
+
+### Zip()
+
+To combine two arrays into an array of pairs, use the [zip()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/zip.html).
+
+```Kotlin
+fun main() { 
+//sampleStart
+    val firstArray = arrayOf(1, 2, 3)
+    val secondArray = arrayOf(1, 2, 3)
+
+    val zipArray = firstArray.zip(secondArray)
+
+    println(zipArray.joinToString(" "))
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="zip-array-kotlin"}
+
+To return a pair of lists from an array of pairs, use the [unzip()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/unzip.html) function.
+
+```Kotlin
+fun main() { 
+//sampleStart
+    val pairArray = arrayOf(Pair(1,1), Pair(2,2), Pair(3,3))
+
+    val unzipArray = pairArray.unzip()
+    println(unzipArray)
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="unzip-array-kotlin"}
+
+## Concatenate arrays
+
+Arrays of type `String` can be concatenated using the `+` operator.
+
+```kotlin
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf("a", "b", "c")
+    val anotherArray = arrayOf("d", "e", "f")
+
+    println((simpleArray + anotherArray).joinToString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="concatenate-array-kotlin"}
+
+## Pass array as vararg
 
-Kotlin also has classes that represent arrays of primitive types without boxing overhead: `ByteArray`,
-`ShortArray`, `IntArray`, and so on. These classes have no inheritance relation to the `Array` class, but they
-have the same set of methods and properties. Each of them also has a corresponding factory function:
+To pass an array to the `vararg` parameter of a function, use the _spread_ operator (`*`).
 
 ```kotlin
-val x: IntArray = intArrayOf(1, 2, 3)
-x[0] = x[1] + x[2]
+val a = arrayOf(1, 2, 3)
+val list = asList(-1, 0, *a, 4)
 ```
 
+For more information, see [Variable number of args (varargs)](functions.md#variable-number-of-arguments-varargs).
+
+## Convert to collection
```

Comment:
Again, check if these headlines are consistent with the way we're usually writing headlines. These phrasings strike me as odd (and even for our very short titles, a little _too_ concise)

Reply:
Can you explain a bit more what part you are finding odd? Would "Convert arrays to collection" sound better to you? Or is it actually that you prefer titles with noun phrases? E.g. Array to collection conversion. 🤔 My preference is to use bare infinitives when it relates to a task. Like Google recommends in their [developer's documentation style guide](https://developers.google.com/style/headings). I checked this with @koshachy  just in case and this also matches with his thinking.